### PR TITLE
[Core] Check all sqlite results and report errors

### PR DESCRIPTION
**Pull request by dmbryson**
This changes debug asserts into runtime error checks. Also changes the
Xcode analyze scheme to use Release mode so that clang surfaces these
warnings.

rdar://problem/40036873

### CODE REVIEW

**Comment by dmbryson** [comment]
FYI, I tested the performance impact of these changes with the PerfTests target and found no measurable difference for the existing tests.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

File: llbuild.xcodeproj/xcshareddata/xcbaselines/E1C404AB1A0308F3003392BA.xcbaseline/80704D72-AD86-42E5-99D3-DB74CDEA1F5C.plist [hunk 1]
```diff
@@ -0,0 +1,171 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dmbryson) [inline llbuild.xcodeproj/xcshareddata/xcbaselines/E1C404AB1A0308F3003392BA.xcbaseline/80704D72-AD86-42E5-99D3-DB74CDEA1F5C.plist]: Hmm, this was not supposed to be added.

Comment (dmbryson) [reply llbuild.xcodeproj/xcshareddata/xcbaselines/E1C404AB1A0308F3003392BA.xcbaseline/80704D72-AD86-42E5-99D3-DB74CDEA1F5C.plist]: Well, it was not intentionally added.  But I do see several other baselines in the repo already.  I'm ambivalent as to whether we should or should not have these here.

Comment (ddunbar) [reply llbuild.xcodeproj/xcshareddata/xcbaselines/E1C404AB1A0308F3003392BA.xcbaseline/80704D72-AD86-42E5-99D3-DB74CDEA1F5C.plist]: Lets not have them, and gitignore them

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**